src/components/product/ProductList.component.tsx

```diff
@@ -1,24 +1,55 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 
-import SearchBar from '@/components/searchbar/SearchBar.component.tsx';
-import data from '@/data/products.json';
-import type { Product } from '@/interfaces/Product';
-
-import ProductCard from './ProductCard.component';
+import { Pagination } from '@/components/product/Pagination.component.tsx';
+import { ProductsComponent } from '@/components/product/Products.component.tsx';
+import { SearchBar } from '@/components/searchbar/SearchBar.component.tsx';
+import type { Product } from '@/interfaces/Product.ts';
 
 import styles from './Product.module.css';
 
-const ProductList = () => (
-    <section className={styles.product__section}>
-        <SearchBar />
-        <div className={styles.container}>
-            <ul className={styles.product__list}>
-                {data.map((product: Product) => (
-                    <ProductCard key={product.id} product={product} />
-                ))}
-            </ul>
-        </div>
-    </section>
-);
-
-export default ProductList;
+export default function ProductList() {
+    const apiUrl = 'https://ma-backend-api.mocintra.com/api/v1/products';
+    const [products, setProducts] = useState<Product[]>([]);
+    const [totalPages, setTotalPages] = useState(0);
+
+    const [paginatedProducts, setPaginatedProducts] = useState<Product[]>([]);
+    const [currentPage, setCurrentPage] = useState<number>(1);
+
+    useEffect(() => {
+        fetch(apiUrl)
+            .then((response) => response.json())
+            .then((data) => {
+                setProducts(data);
+                const paginated = getPaginatedProducts(data);
+                setPaginatedProducts(paginated);
+
+                setTotalPages(Math.ceil(data.length / 8));
+            });
+    });
```

Comment:
Please add a dependecies - now useEffect makes a request on every rerender and DDOSes our backend)))
![image](https://github.com/demasha0908/fe-react-2024/assets/98031768/b0f33e22-d600-4991-adbf-a185c7303b39)